api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -40,9 +40,7 @@
    *
    * @return this catalog's name
    */
-  default String name() {
-    return toString();
-  }
+  String name();
```

Comment:
This breaks the public API. Is there a way to handle it differently? Can we maybe just drop `name` from the default implementation of `toString` in the base metastore catalog? Or any other option that keeps the old API.

Reply:
make sense!